app.py

```diff
@@ -28,6 +31,36 @@
 ]  # "Hashstack V0", "Hashstack V1"
 
 
+
+    
+def fetch_token_symbols2(token_addresses: Set[str]) -> Dict[str, str]:
+    async def async_fetch():
+        return {addr: await get_underlying_token_symbol(addr) for addr in token_addresses}
+    return asyncio.run(async_fetch())
+
+async def get_underlying_token_symbol(token_address: str) -> str:
+    try:
+        underlying_token_address = await src.blockchain_call.func_call(
+            addr=token_address,
+            selector="underlyingAsset",
+            calldata=[],
+        )
+        underlying_token_address = src.helpers.add_leading_zeros(
+            hex(underlying_token_address[0])
+        )
+        underlying_token_symbol = await src.helpers.get_symbol(
+            token_address=underlying_token_address
+        )
+        return underlying_token_symbol.strip('\x00')
```

Comment:
Please, move it to `constant` variable `'\x00'`
In python it's call magic string read more here: https://softwareengineering.stackexchange.com/questions/365339/what-is-wrong-with-magic-strings